### ISSUE TITLE
Firefox driver doesn't wait for page loads

### ISSUE THREAD

**Issue by CROSP** [mozilla/geckodriver#735]
## Firefox Version
v0.16

## Platform
Windows/OSX/Linux - Everywhere

## Steps to reproduce -
I am having troubles with the Gecko driver when using Selenium.

Here is my simple test script written in Python

```
driver = webdriver.Remote(
    command_executor='http://127.0.0.1:4444/wd/hub',
    desired_capabilities=DesiredCapabilities.FIREFOX)
driver.get('http://mypage.com')
selectLocation = Select(driver.find_element_by_id("login-page"))
selectLocation.select_by_visible_text('User Area')
#time.sleep(5)
loginInput = driver.find_element_by_name("username")
loginInput.send_keys("hello")
loginInput = driver.find_element_by_name("password")
loginInput.send_keys("world")
driver.quit()
```
Firefox webdriver doesn't wait until page is loaded and tries to find elements without waiting. 
This flow works perfectly with the Chrome driver.
If I uncomment the sleep statement, it will wait and then being able to find the element. 

I am running Selenium inside a docker container and I've tried different versions of Firefox containers, but the behavior is the same.

Furthermore I have tried different languages `Python`, `Java`. `NodeJS` . And also tried to run the native driver (not remote) on my local machine. The same behavior. 

Is it possible to fix this issue ? It doesn't make any sense to insert wait statements everywhere.

**Comment by jrbenny35** [comment]
What version of Firefox?

**Comment by CROSP** [comment]
Firefox version in the docker container is **53.0**.  Locally on my machine I have **53.0.2 (32-bit)**.

**Comment by jrbenny35** [comment]
From what I know the wait for page patch is only in version 55 currently.

**Comment by CROSP** [comment]
@jrbenny35  Where I can get it ?)) The latest available version on the official page is 53.0.2. Please share the  source of this information.

**Comment by AutomatedTester** [comment]
@jrbenny35 that patch is unrelated to the example given.

Since there is no log, which is asked for in the issue template, and the example given isnt against a proper website I am left to guess. I am guessing the select causes a page load meaning that it is a click and wait.

This has been fixed in https://bugzilla.mozilla.org/show_bug.cgi?id=1335778 and if you want to try with Firefox Nightly you will be able to see if it now works. If it doesn't we will need a tracelog from GeckoDriver and a _working_ example with html that we can use to reproduce.
